x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/task/ReindexDataStreamPersistentTaskExecutor.java

```diff
@@ -108,6 +108,9 @@ protected void nodeOperation(
         ReindexDataStreamTaskParams params,
         PersistentTaskState persistentTaskState
     ) {
+        if (isComplete(persistentTaskState)) {
```

Comment:
Don't you have to schedule the `completeTask` for removal here, so the task is cleaned up after 24h?

Reply:
Ah, good point -- the scheduling of the removal was on the threadpool on the old node, which won't carry over to this node. But I think I can just do `task.markAsCompleted();` here. The user has already upgraded, meaning they know that it completed. And the information is really out of date at this point anyway (because the data stream is "old" for this new cluster).

Reply:
If a node is terminated and the task moves to a new node (after completion), we wouldn't remove the task metadata otherwise. 

Also note, the `ScheduledCancellable` returned by below line keeps running on the old node in this case, making the behavior a bit indeterministic. As far as I remember `completeAndNotifyIfNeeded` might throw if completed multiple times in some cases.
```
threadPool.schedule(completeTask, timeToLive, threadPool.generic())
```

Reply:
I would guess that the task would only be moved if the node it was on came down altogether, threadpool included, right? Regardless though, the exception will be harmless won't it?
I'll change it to schedule the removal in the future so that it continues to work once users intentionally run this on 9.x (I don't think they will want to until we're approaching 10.0).

Reply:
> The user has already upgraded, meaning they know that it completed. And the information is really out of date at this point anyway (because the data stream is "old" for this new cluster).

Actually, rethinking this, the task might also be moved for other reasons (a node might crash any time), we can't assume the upgrade has happened just because the task was moved. Not sure how important it is to keep the old state around for 24h, but it might lead to surprising results this way.

Reply:
Yeah this change definitely doesn't assume the task was moved just because it's on a new node -- it is explicitly checking that the completion time is non-null, meaning it was completed on some other node.

Reply:
It only keeps the task for 24h beyond whenever it completes. If a node crashes before it completes, it will not have been scheduled for removal.

Reply:
Yes, this is only ever a problem if a node crashes after completion within those 24h

Reply:
👍 looks like that's already addressed